base/compiler/tfuncs.jl

```diff
@@ -1648,7 +1648,7 @@ function return_type_tfunc(interp::AbstractInterpreter, argtypes::Vector{Any}, s
                     if contains_is(argtypes_vec, Union{})
                         return Const(Union{})
                     end
-                    rt = abstract_call(interp, nothing, argtypes_vec, sv, -1).rt
+                    rt = widenconditional(abstract_call(interp, nothing, argtypes_vec, sv, -1).rt)
```

Comment:
Now this change isn't necessary for this PR, but I think `widenconditional` is better here since it may produce more accurate result since it can be widen to `Const`, so let me include this as is.